src/test/java/kitchenpos/application/ProductServiceTest.java

```diff
@@ -0,0 +1,221 @@
+                testProductRepository,
+                testMenuRepository,
+                (name) -> false
+        );
+        BigDecimal price = BigDecimal.valueOf(10);
+        String name = "testName";
+        Product request = new Product();
+        request.setPrice(price);
+        request.setName(name);
+        Product product = productService.create(request);
+
+        List<MenuProduct> menuProducts = List.of(createMenuProduct(product, 1L));
+        Menu menu = createMenu(BigDecimal.valueOf(5), menuProducts);
+        testMenuRepository.save(menu);
+
+
+
+        // when
+        Product changeProductRequest = new Product();
+        BigDecimal newPrice = BigDecimal.valueOf(1);
+        changeProductRequest.setPrice(newPrice);
+        productService.changePrice(product.getId(), changeProductRequest);
+
+        // then
+        assertAll(
+                () -> assertThat(testProductRepository.findById(product.getId()).get().getPrice()).isEqualTo(newPrice),
+                () -> assertFalse(testMenuRepository.findById(menu.getId()).get().isDisplayed())
+        );
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(longs = {-1L})
+    @DisplayName("가격은 존재하며 0이상이다")
+    void create_change_product_price_fail(Long invalidPrice) {
+        // given
+        MenuRepository testMenuRepository = new TestMenuRepository();
+        ProductRepository testProductRepository = new TestProductRepository();
+        ProductService productService = new ProductService(
+                testProductRepository,
+                testMenuRepository,
+                (name) -> false
+        );
+        BigDecimal price = BigDecimal.valueOf(10);
+        String name = "testName";
+        Product request = new Product();
+        request.setPrice(price);
+        request.setName(name);
+        Product product = productService.create(request);
+
+
+        // when
+        Product changeProductRequest = new Product();
+        BigDecimal newPrice = invalidPrice == null ? null : BigDecimal.valueOf(invalidPrice);
+        changeProductRequest.setPrice(newPrice);
+        assertThrows(IllegalArgumentException.class, () ->productService.changePrice(product.getId(), changeProductRequest));
+    }
+
+    @Test
+    @DisplayName("모든 제품 조회")
+    void find_all() {
+        ProductService productService = new ProductService(
+                new TestProductRepository(),
+                new TestMenuRepository(),
+                (name) -> false
+        );
+        BigDecimal price = BigDecimal.valueOf(10);
+        String name = "testName";
+        Product request = new Product();
+        request.setPrice(price);
+        request.setName(name);
+        Product product = productService.create(request);
+
+        List<Product> products = productService.findAll();
+        assertAll(
+                () -> assertThat(products.size()).isEqualTo(1),
+                () -> assertThat(products.get(0).getId()).isEqualTo(product.getId())
```

Comment:
사소한 팁인데 java 21부터 List에 getFirst()라는 편의 메서드가 생겼어요.
```suggestion
                () -> assertThat(products.getFirst().getId()).isEqualTo(product.getId())
```

Reply:
감사합니다! 적용했습니다